src/backend/web.rs

```diff
@@ -69,6 +69,76 @@ impl<F: Future> Future for MakeSendFuture<F> {
 
 unsafe impl<F> Send for MakeSendFuture<F> {}
 
+fn future_map<Fut, F, O>(fut: Fut, f: F) -> Map<Fut, F>
+where
+    Fut: Future,
+    F: FnOnce(Fut::Output) -> O,
+{
+    Map::new(fut, f)
+}
+
+pin_project! {
```

Comment:
Could we avoid `pin_project` here too? I think we should be able to do something simpler here like we have for `MakeSendFuture` (possibly even collapsing them together)

Reply:
We can, but `pin-project-lite` is already a dependency of `naga`, so I am not sure what we gain by making that change. Merging with `MakeSendFuture` sounds like a good idea!

Reply:
I'm curious as to where Naga would need it.
Anyway, Naga is not a wgpu-rs dependency for the `wasm32` target.

Reply:
I misspoke. not Naga. But `tracing`:

```
tracing v0.1.22
├── cfg-if v1.0.0
├── pin-project-lite v0.2.0
├── tracing-attributes v0.1.11 (proc-macro)
│   ├── proc-macro2 v1.0.24
│   │   └── unicode-xid v0.2.1
│   ├── quote v1.0.7
│   │   └── proc-macro2 v1.0.24 (*)
│   └── syn v1.0.54
│       ├── proc-macro2 v1.0.24 (*)
│       ├── quote v1.0.7 (*)
│       └── unicode-xid v0.2.1
└── tracing-core v0.1.17
    └── lazy_static v1.4.0
```

Reply:
Alright :+1: I think ideally tracing shouldn't be a dependency for wasm (at least by default), so removing the dependency would still be useful to keep wasm builds smaller. But we could probably tackle this in a follow-up PR instead

Reply:
Agreed, we don't need to log anything with tracing when targeting the Web, but can follow-up.
Although, now, instead of just cfg-blocking `tracing` usage, we'd have to figure out how to rewrite the pin section...
So it would be nicer to deal with it right away.